src/Renderer/AbstractRenderer.php

```diff
@@ -124,6 +166,28 @@ protected function _background_image($url, $x, $y, $width, $height, $style)
         $bg_width = round((float)($width * $dpi) / 72);
         $bg_height = round((float)($height * $dpi) / 72);
 
+        $_background_size = $style->get_background_size();
+        $is_calculated_background = ! is_array($_background_size);
+        list($img_w, $img_h) = $this->_resize_background_image(
+            $img_w,
+            $img_h,
+            $bg_width,
+            $bg_height,
+            $_background_size
+        );
+
+        if (Helpers::is_percent($img_w)) {
```

Comment:
One of the calculations we need to accommodate, in addition to a percentage and a unit value, is when one of the sides is set to "auto". In that case the side specified as auto should be calculated based on the ratio compared to the length of the other side. If both sides are auto we just use the image dimensions.

Reply:
Auto handling / calculation implemented now